contracts/UnionZap.sol

```diff
@@ -49,22 +51,33 @@ contract UnionZap is Ownable, UnionBase {
         bytes32[] merkleProof;
     }
 
+    struct tokenContracts {
+        address pool;
+        address swapper;
+        address distributor;
+    }
+
     struct curveSwapParams {
         address pool;
         uint16 ethIndex;
     }
 
+    mapping(address => tokenContracts) public tokenInfo;
     mapping(address => curveSwapParams) public curveRegistry;
 
     event Received(address sender, uint256 amount);
-    event Distributed(uint256 amount, uint256 fees, bool locked);
-    event DistributorUpdated(address distributor);
+    event Distributed(uint256 amount, address token, address distributor);
     event VotiumDistributorUpdated(address distributor);
     event FundsRetrieved(address token, address to, uint256 amount);
     event CurvePoolUpdated(address token, address pool);
+    event OutputTokenUpdated(
+        address token,
+        address pool,
+        address swapper,
+        address distributor
+    );
 
     constructor(address distributor_) {
```

Comment:
Just realized distributooor is unused here

Reply:
Fixed in ed9c99347be53170feeba71c70a3641be2b28fe4